Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.cpp

```diff
@@ -0,0 +1,182 @@
+/*==LICENSE==*
+
+CyanWorlds.com Engine - MMOG client, server and tools
+Copyright (C) 2011  Cyan Worlds, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#include "plSrtFileReader.h"
+
+#include "hsStream.h"
+#include "plStatusLog/plStatusLog.h"
+
+#include <regex>
+
+static const std::regex speakerTextRegex("^((([\\w.]+(\\s\\w+)?):) )?(.*)");
+static const std::regex timingsRegex("^(\\d{2}):(\\d{2}):(\\d{2}),(\\d{3}) --> (\\d{2}):(\\d{2}):(\\d{2}),(\\d{3})$");
+
+bool plSrtFileReader::ReadFile()
+{
+    plFileName audioSrtPath = plFileName::Join(plFileSystem::GetCWD(), "dat", fAudioFileName.StripFileExt() + ".srt");
```

Comment:
Have you given any thought to the suggestion I relayed from @Deledrius about using a format agnostic extension? Also, I think that, for consistency with Myst V, the subtitles should go in the sfx directory.

Reply:
Myst V's subtitles are actually not in the sfx folder. The `.sub` files are in dat, there are `.sgt` files in sfx that are something else.

Reply:
Whoops. Regardless, I think that if we're matching the audio files 1:1, these need to go in the sfx folder.

Reply:
I've thought about it, but does anyone have an agnostic extension they wish to put forward as an option? Otherwise I'll lean toward something silly, self-aggrandizing and seemingly-unused by other applications like `.yez` ;) As for the folder, I can go ahead and move them if we have agreement?

Reply:
Myst V uses `.sub` which seems suitably generic, but I don't see a problem with `.srt` if that's the standard extension that's well-supported by existing tools.

Reply:
Since I didn't specify, @Deledrius's suggestion was to use an agnostic extension so the format could be changed later (without leaving stale files in the client folder) if the srt format is determined to be insufficient in the future.

Reply:
Ah, I thought `.sub` was for a specific format of subtitles like `.srt` was, but it looks from The Googles that it can refer to subtitles in many different unspecified formats, so I'm good with that. I think Deledrius suggested an agnostic extension so they could be changed to a different format internally more easily without needing re...manifesting, I guess?

Reply:
I honestly used Notepad++ and a regular audio player app because the format is dirt simple to use, but I tried https://www.happyscribe.com/subtitle-tools/online-subtitle-editor/free out a little bit after I finished the majority of the SRTs and it seems OK if you prefer a more structured way to edit the SRT file. It reads and writes .srt files just fine, and did let me upload the `.sub` file without issue (after selecting "All Files" in the file browser). That tool, and the majority of free online tooling, assumes you're subtitling video, though, so you'd still have to use a separate audio player.